File: app/src/main/java/me/toptas/fancyshowcasesample/BaseActivity.java
package me.toptas.fancyshowcasesample;

import android.support.v7.app.AppCompatActivity;

import me.toptas.fancyshowcase.FancyShowCaseView;

/**
 * Created by yeldos on 3/29/17.
 */

public abstract class BaseActivity extends AppCompatActivity {

    @Override
    public void onBackPressed() {
        if (FancyShowCaseView.isVisible(this)) {
            FancyShowCaseView.hidePlease(this);
        } else {
            super.onBackPressed();
        }
    }
}

File: app/src/main/java/me/toptas/fancyshowcasesample/BaseActivity.kt
package me.toptas.fancyshowcasesample

import android.support.v7.app.AppCompatActivity

import me.toptas.fancyshowcase.FancyShowCaseView

/**
 * Created by yeldos on 3/29/17.
 */

abstract class BaseActivity : AppCompatActivity() {

    override fun onBackPressed() {
        if (FancyShowCaseView.isVisible(this)) {
            FancyShowCaseView.hideCurrent(this)
        } else {
            super.onBackPressed()
        }
    }
}
